PluginsAndFeatures/azure-toolkit-for-eclipse/com.microsoft.azuretools.hdinsight/src/com/microsoft/azuretools/hdinsight/spark/common2/SparkSubmitHelper.java

```diff
@@ -179,6 +179,8 @@ public void printRunningLogStreamingly(/*Project project,*/ int id, IClusterDeta
 
             if (isFailedJob) {
                 postEventProperty.put("IsRunningSucceed", "false");
+                HttpResponse httpResponse = SparkBatchSubmission.getInstance().getBatchJobFullLog(clusterDetail.getConnectionUrl() + "/livy/batches", id);
+                postEventProperty.put("SubmitFailedReason", httpResponse.getMessage().substring(0, 50));
```

Comment:
Do we need to get a full log here? Could we use some info in `SparkSubmitResponse`?
And seems the first 50 chars of the full log is hard to contain useful information for failed job.